File: LeetCode_35/main.cpp
/**
    LeetCode_35
    https://leetcode.com/problems/search-insert-position/description/
    MatrixDeity, 2017.
*/

#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {
        if (nums.empty())
            return 0;

        int i = 0, j = (int) nums.size() - 1;
        while (i <= j) {
            int m = i + (j - i) / 2;
            if (target > nums[m])
                i = m + 1;
            else
                j = m - 1;
        }
        return i;
    }
};

int main() {
    return 0;
}

File: LeetCode_35/main.go
/**
  LeetCode_35
  https://leetcode.com/problems/search-insert-position/
  MatrixDeity, 2023.
*/

package leetcode35

func searchInsert(nums []int, target int) int {
	i, j := 0, len(nums)-1

	for i <= j {
		m := i + (j-i)/2
		if nums[m] < target {
			i = m + 1
		} else {
			j = m - 1
		}
	}

	return i
}
